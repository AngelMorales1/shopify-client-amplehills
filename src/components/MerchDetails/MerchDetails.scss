@import 'settings';

.MerchDetails {
  padding: spacing(4) spacing(6);

  @media ($medium-up) {
    padding: spacing(4) spacing(16);
  }

  &__breadcrumbs {
    margin-left: spacing(-4);
  }

  &__container {
    flex-direction: column;

    @media ($medium-up) {
      flex-direction: row-reverse;
      justify-content: space-between;
    }
  }

  &__order-container {
    @media ($medium-up) {
      padding: 0 spacing(20);
    }
  }

  &__image,
  &__image--divide {
    border-radius: radius('s');
  }

  &__image-container--divide {
    flex-direction: column;
    margin-bottom: spacing(2);

    @media ($medium-up) {
      flex-direction: row;
      margin: 0;
    }

    .MerchDetails__image--divide:first-child {
      margin-bottom: spacing(4);

      @media ($medium-up) {
        margin: 0;
        margin-right: spacing(2);
      }
    }

    .MerchDetails__image--divide:last-child {
      @media ($medium-up) {
        margin-left: spacing(2);
      }
    }
  }

  &__image--divide {
    width: 100%;

    @media ($medium-up) {
      width: 50%;
    }
  }
}
