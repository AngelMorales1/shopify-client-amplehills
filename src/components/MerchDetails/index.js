import React, { Component, Fragment } from 'react';
import PropTypes from 'prop-types';
import marked from 'marked';
import cx from 'classnames';
import get from 'utils/get';
import contentfulImgUtil from 'utils/contentfulImgUtil';
import merchModel from 'models/merchModel';
import checkoutModel from 'models/checkoutModel';
import Global from 'constants/Global';

import { Button, QuantitySelector, Radio } from 'components/base';
import Breadcrumbs from 'components/Breadcrumbs';
import styles from './MerchDetails.scss';

class MerchDetails extends Component {
  state = {
    currentBreakpoint: Global.breakpoints.small.label,
    selectedItem: '',
    quantity: 1
  };

  componentDidMount() {console.log(get(this, 'props', ''))
    const firstAvailableItem = get(this, 'props.merch.variants', '').find(
      variant => variant.available === true
    );
    this.setState({ selectedItem: firstAvailableItem.id });
    window.addEventListener('resize', this.updateWindow);

    this.updateWindow();
  }

  componentWillUnmount() {
    window.removeEventListener('resize', this.updateWindow);
    window.removeEventListener('scroll', this.updateMenu);
  }

  updateWindow = () => {
    const { small, large } = Global.breakpoints;
    const currentBreakpoint =
      window.innerWidth <= large.lowerbound ? small.label : large.label;

    if (this.state.currentBreakpoint !== currentBreakpoint) {
      this.setState({ currentBreakpoint });
    }
  };

  addToCart = () => {
    const selectedItem = get(this, 'props.merch.variants', []).find(
      variant => variant.id === this.state.selectedItem
    );

    const item = [
      {
        variantId: get(selectedItem, 'id', ''),
        quantity: this.state.quantity,
        customAttributes: [
          {
            key: 'Item',
            value: get(selectedItem, 'title', '')
          }
        ]
      }
    ];

    this.props.actions.addLineItems(this.props.checkout.id, item);
  };

  render() {
    const merch = get(this, 'props.merch', {});
    const breadcrumbs = [{ to: '/order-online', label: 'Order Online' }];
    const images = get(merch, 'images', []);
    const selectedItem = get(merch, 'variants', []).find(
      variant => variant.id === this.state.selectedItem
    );

    return (
      <div className={cx(styles['MerchDetails'])}>
        <Breadcrumbs
          className={cx(styles['MerchDetails__breadcrumbs'])}
          breadcrumbs={breadcrumbs}
        />
        <div className={cx(styles['MerchDetails__container'], 'flex')}>
          <div
            className={cx(
              styles['MerchDetails__image'],
              'aspect-4-3 my2 col-12 md-hide lg-hide'
            )}
            style={{
              background: `url(${contentfulImgUtil(
                get(merch, 'images[0].fields.file.url', ''),
                '900'
              )}) no-repeat center`,
              backgroundSize: 'cover'
            }}
          />
          <div
            className={cx(
              styles['MerchDetails__order-container'],
              'col-12 md-col-6 my2 flex flex-column items-center'
            )}
          >
            <div className="mx-auto text-container-width">
              <h2 className="block-headline mb3">{merch.title}</h2>
              <div
                dangerouslySetInnerHTML={{
                  __html: marked(get(merch, 'description', ''))
                }}
                className="markdown-block"
              />
              <div className="flex flex-row flex-wrap">
                {get(merch, 'variants', []).map(variant => (
                  <Radio
                    disabled={!variant.available}
                    color={variant.available ? 'peach' : 'ghost-gray'}
                    key={variant.id}
                    checked={this.state.selectedItem === variant.id}
                    onClick={() => this.setState({ selectedItem: variant.id })}
                    className="mr3 my2 small"
                    label={variant.title}
                  />
                ))}
              </div>
              <div className="flex flex-row flex-wrap justify-between items-center mt3 mb4">
                <QuantitySelector
                  variant={
                    this.state.currentBreakpoint === 'small' ? 'small' : null
                  }
                  className="my3 mr2"
                  quantity={this.state.quantity}
                  onChange={value => this.setState({ quantity: value })}
                />
                <Button
                  className={cx(styles['MerchDetails__button'])}
                  color="madison-blue"
                  variant={
                    this.state.currentBreakpoint === 'small'
                      ? 'primary-small'
                      : 'primary'
                  }
                  shadow={true}
                  onClick={this.addToCart}
                >
                  <span>Add to Cart</span>
                  <span className="ml2">
                    ${(
                      get(selectedItem, 'price', 0.0) * this.state.quantity
                    ).toFixed(2)}
                  </span>
                </Button>
              </div>
              <p className="bold uppercase text-peach mb2">
                {merch.detailsTitle}
              </p>
              <div
                dangerouslySetInnerHTML={{
                  __html: marked(get(merch, 'detailsContent', ''))
                }}
                className="markdown-block"
              />
            </div>
          </div>
          <div className="col-12 md-col-6">
            <div
              className={cx(
                styles['MerchDetails__image'],
                'aspect-4-3 my2 xs-hide sm-hide'
              )}
              style={{
                background: `url(${contentfulImgUtil(
                  get(images[0], 'fields.file.url', ''),
                  '900'
                )}) no-repeat center`,
                backgroundSize: 'cover'
              }}
            />
            {images.length > 4 ? (
              <Fragment>
                <div
                  className={cx(
                    styles['MerchDetails__image-container--divide'],
                    'w100 flex'
                  )}
                >
                  <div
                    className={cx(
                      styles['MerchDetails__image--divide'],
                      'aspect-4-3'
                    )}
                    style={{
                      background: `url(${contentfulImgUtil(
                        get(images[1], 'fields.file.url', ''),
                        '900'
                      )}) no-repeat center`,
                      backgroundSize: 'cover'
                    }}
                  />
                  <div
                    className={cx(
                      styles['MerchDetails__image--divide'],
                      'aspect-4-3'
                    )}
                    style={{
                      background: `url(${contentfulImgUtil(
                        get(images[2], 'fields.file.url', ''),
                        '900'
                      )}) no-repeat center`,
                      backgroundSize: 'cover'
                    }}
                  />
                </div>
                {images.slice(3).map((image, i) => (
                  <div
                    key={get(image, 'sys.id', i)}
                    className={cx(
                      styles['MerchDetails__image'],
                      'aspect-4-3 my2'
                    )}
                    style={{
                      background: `url(${contentfulImgUtil(
                        get(image, 'fields.file.url', ''),
                        '900'
                      )}) no-repeat center`,
                      backgroundSize: 'cover'
                    }}
                  />
                ))}
              </Fragment>
            ) : (
              <Fragment>
                {images.slice(1).map((image, i) => (
                  <div
                    key={get(image, 'sys.id', i)}
                    className={cx(
                      styles['MerchDetails__image'],
                      'aspect-4-3 my2'
                    )}
                    style={{
                      background: `url(${contentfulImgUtil(
                        get(image, 'fields.file.url', ''),
                        '900'
                      )}) no-repeat center`,
                      backgroundSize: 'cover'
                    }}
                  />
                ))}
              </Fragment>
            )}
          </div>
        </div>
      </div>
    );
  }
}

MerchDetails.propTypes = {
  merch: merchModel.propTypes,
  actions: PropTypes.shape({
    addLineItems: PropTypes.func
  }),
  checkout: checkoutModel.propTypes
};

MerchDetails.defaultProps = {
  merch: merchModel.default,
  actions: {
    addLineItems: () => {}
  },
  checkout: checkoutModel.default
};

export default MerchDetails;
