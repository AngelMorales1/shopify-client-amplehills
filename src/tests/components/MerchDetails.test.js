import React from 'react';
import { shallow } from 'enzyme';

import MerchDetails from 'components/MerchDetails';

it('renders without data', () => {
  const component = shallow(<MerchDetails />);
  expect(component).toMatchSnapshot();
});
it('renders with data', () => {
  const component = shallow(
    <MerchDetails
      merch={{
        available: true,
        description: 'foo',
        detailsContent: 'foo',
        detailsTitle: 'foo',
        handle: 'foo',
        id: 'foo',
        images: [],
        price: 0,
        title: '',
        variants: []
      }}
    />
  );
  expect(component).toMatchSnapshot();
});
