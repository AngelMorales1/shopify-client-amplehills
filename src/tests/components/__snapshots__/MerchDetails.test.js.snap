// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with data 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MerchDetails
    merch={
        Object {
            "available": true,
            "description": "foo",
            "detailsContent": "foo",
            "detailsTitle": "foo",
            "handle": "foo",
            "id": "foo",
            "images": Array [],
            "price": 0,
            "title": "",
            "variants": Array [],
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Breadcrumbs
          breadcrumbs={
                    Array [
                              Object {
                                "label": "Order Online",
                                "to": "/order-online",
                              },
                            ]
          }
          className=""
/>,
        <div
          className="flex"
>
          <div
                    className="aspect-4-3 my2 col-12 md-hide lg-hide"
                    style={
                              Object {
                                        "background": "url(?w=900&fm=jpg&fl=progressive&q=90) no-repeat center",
                                        "backgroundSize": "cover",
                                      }
                    }
          />
          <div>
                    details
          </div>
          <div
                    className="col-12 md-col-6"
          >
                    <div
                              className="aspect-4-3 my2 xs-hide sm-hide"
                              style={
                                        Object {
                                                  "background": "url(?w=900&fm=jpg&fl=progressive&q=90) no-repeat center",
                                                  "backgroundSize": "cover",
                                                }
                              }
                    />
                    <Unknown>
                              
                    </Unknown>
          </div>
</div>,
      ],
      "className": "",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "breadcrumbs": Array [
            Object {
              "label": "Order Online",
              "to": "/order-online",
            },
          ],
          "className": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="aspect-4-3 my2 col-12 md-hide lg-hide"
              style={
                            Object {
                                          "background": "url(?w=900&fm=jpg&fl=progressive&q=90) no-repeat center",
                                          "backgroundSize": "cover",
                                        }
              }
/>,
            <div>
              details
</div>,
            <div
              className="col-12 md-col-6"
>
              <div
                            className="aspect-4-3 my2 xs-hide sm-hide"
                            style={
                                          Object {
                                                        "background": "url(?w=900&fm=jpg&fl=progressive&q=90) no-repeat center",
                                                        "backgroundSize": "cover",
                                                      }
                            }
              />
              <Unknown>
                            
              </Unknown>
</div>,
          ],
          "className": "flex",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "aspect-4-3 my2 col-12 md-hide lg-hide",
              "style": Object {
                "background": "url(?w=900&fm=jpg&fl=progressive&q=90) no-repeat center",
                "backgroundSize": "cover",
              },
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "details",
            },
            "ref": null,
            "rendered": "details",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="aspect-4-3 my2 xs-hide sm-hide"
                  style={
                                    Object {
                                                      "background": "url(?w=900&fm=jpg&fl=progressive&q=90) no-repeat center",
                                                      "backgroundSize": "cover",
                                                    }
                  }
/>,
                <Unknown>
                  
</Unknown>,
              ],
              "className": "col-12 md-col-6",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "aspect-4-3 my2 xs-hide sm-hide",
                  "style": Object {
                    "background": "url(?w=900&fm=jpg&fl=progressive&q=90) no-repeat center",
                    "backgroundSize": "cover",
                  },
                },
                "ref": null,
                "rendered": null,
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": Array [],
                },
                "ref": null,
                "rendered": Array [],
                "type": Symbol(react.fragment),
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Breadcrumbs
            breadcrumbs={
                        Array [
                                    Object {
                                      "label": "Order Online",
                                      "to": "/order-online",
                                    },
                                  ]
            }
            className=""
/>,
          <div
            className="flex"
>
            <div
                        className="aspect-4-3 my2 col-12 md-hide lg-hide"
                        style={
                                    Object {
                                                "background": "url(?w=900&fm=jpg&fl=progressive&q=90) no-repeat center",
                                                "backgroundSize": "cover",
                                              }
                        }
            />
            <div>
                        details
            </div>
            <div
                        className="col-12 md-col-6"
            >
                        <div
                                    className="aspect-4-3 my2 xs-hide sm-hide"
                                    style={
                                                Object {
                                                            "background": "url(?w=900&fm=jpg&fl=progressive&q=90) no-repeat center",
                                                            "backgroundSize": "cover",
                                                          }
                                    }
                        />
                        <Unknown>
                                    
                        </Unknown>
            </div>
</div>,
        ],
        "className": "",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "breadcrumbs": Array [
              Object {
                "label": "Order Online",
                "to": "/order-online",
              },
            ],
            "className": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="aspect-4-3 my2 col-12 md-hide lg-hide"
                style={
                                Object {
                                                "background": "url(?w=900&fm=jpg&fl=progressive&q=90) no-repeat center",
                                                "backgroundSize": "cover",
                                              }
                }
/>,
              <div>
                details
</div>,
              <div
                className="col-12 md-col-6"
>
                <div
                                className="aspect-4-3 my2 xs-hide sm-hide"
                                style={
                                                Object {
                                                                "background": "url(?w=900&fm=jpg&fl=progressive&q=90) no-repeat center",
                                                                "backgroundSize": "cover",
                                                              }
                                }
                />
                <Unknown>
                                
                </Unknown>
</div>,
            ],
            "className": "flex",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "aspect-4-3 my2 col-12 md-hide lg-hide",
                "style": Object {
                  "background": "url(?w=900&fm=jpg&fl=progressive&q=90) no-repeat center",
                  "backgroundSize": "cover",
                },
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "details",
              },
              "ref": null,
              "rendered": "details",
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <div
                    className="aspect-4-3 my2 xs-hide sm-hide"
                    style={
                                        Object {
                                                            "background": "url(?w=900&fm=jpg&fl=progressive&q=90) no-repeat center",
                                                            "backgroundSize": "cover",
                                                          }
                    }
/>,
                  <Unknown>
                    
</Unknown>,
                ],
                "className": "col-12 md-col-6",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "aspect-4-3 my2 xs-hide sm-hide",
                    "style": Object {
                      "background": "url(?w=900&fm=jpg&fl=progressive&q=90) no-repeat center",
                      "backgroundSize": "cover",
                    },
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "children": Array [],
                  },
                  "ref": null,
                  "rendered": Array [],
                  "type": Symbol(react.fragment),
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`renders without data 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MerchDetails
    merch={
        Object {
            "available": false,
            "description": "",
            "detailsContent": "",
            "detailsTitle": "",
            "handle": "",
            "id": "",
            "images": Array [
              Object {
                "fields": Object {
                  "file": Object {
                    "url": "",
                  },
                  "title": "",
                },
                "sys": Object {
                  "id": "",
                },
              },
            ],
            "price": 0,
            "title": "",
            "variants": Array [
              Object {
                "available": false,
                "id": "",
                "price": 0,
                "title": "",
              },
            ],
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Breadcrumbs
          breadcrumbs={
                    Array [
                              Object {
                                "label": "Order Online",
                                "to": "/order-online",
                              },
                            ]
          }
          className=""
/>,
        <div
          className="flex"
>
          <div
                    className="aspect-4-3 my2 col-12 md-hide lg-hide"
                    style={
                              Object {
                                        "background": "url(?w=900&fm=jpg&fl=progressive&q=90) no-repeat center",
                                        "backgroundSize": "cover",
                                      }
                    }
          />
          <div>
                    details
          </div>
          <div
                    className="col-12 md-col-6"
          >
                    <div
                              className="aspect-4-3 my2 xs-hide sm-hide"
                              style={
                                        Object {
                                                  "background": "url(?w=900&fm=jpg&fl=progressive&q=90) no-repeat center",
                                                  "backgroundSize": "cover",
                                                }
                              }
                    />
                    <Unknown>
                              
                    </Unknown>
          </div>
</div>,
      ],
      "className": "",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "breadcrumbs": Array [
            Object {
              "label": "Order Online",
              "to": "/order-online",
            },
          ],
          "className": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="aspect-4-3 my2 col-12 md-hide lg-hide"
              style={
                            Object {
                                          "background": "url(?w=900&fm=jpg&fl=progressive&q=90) no-repeat center",
                                          "backgroundSize": "cover",
                                        }
              }
/>,
            <div>
              details
</div>,
            <div
              className="col-12 md-col-6"
>
              <div
                            className="aspect-4-3 my2 xs-hide sm-hide"
                            style={
                                          Object {
                                                        "background": "url(?w=900&fm=jpg&fl=progressive&q=90) no-repeat center",
                                                        "backgroundSize": "cover",
                                                      }
                            }
              />
              <Unknown>
                            
              </Unknown>
</div>,
          ],
          "className": "flex",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "aspect-4-3 my2 col-12 md-hide lg-hide",
              "style": Object {
                "background": "url(?w=900&fm=jpg&fl=progressive&q=90) no-repeat center",
                "backgroundSize": "cover",
              },
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "details",
            },
            "ref": null,
            "rendered": "details",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="aspect-4-3 my2 xs-hide sm-hide"
                  style={
                                    Object {
                                                      "background": "url(?w=900&fm=jpg&fl=progressive&q=90) no-repeat center",
                                                      "backgroundSize": "cover",
                                                    }
                  }
/>,
                <Unknown>
                  
</Unknown>,
              ],
              "className": "col-12 md-col-6",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "aspect-4-3 my2 xs-hide sm-hide",
                  "style": Object {
                    "background": "url(?w=900&fm=jpg&fl=progressive&q=90) no-repeat center",
                    "backgroundSize": "cover",
                  },
                },
                "ref": null,
                "rendered": null,
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": Array [],
                },
                "ref": null,
                "rendered": Array [],
                "type": Symbol(react.fragment),
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Breadcrumbs
            breadcrumbs={
                        Array [
                                    Object {
                                      "label": "Order Online",
                                      "to": "/order-online",
                                    },
                                  ]
            }
            className=""
/>,
          <div
            className="flex"
>
            <div
                        className="aspect-4-3 my2 col-12 md-hide lg-hide"
                        style={
                                    Object {
                                                "background": "url(?w=900&fm=jpg&fl=progressive&q=90) no-repeat center",
                                                "backgroundSize": "cover",
                                              }
                        }
            />
            <div>
                        details
            </div>
            <div
                        className="col-12 md-col-6"
            >
                        <div
                                    className="aspect-4-3 my2 xs-hide sm-hide"
                                    style={
                                                Object {
                                                            "background": "url(?w=900&fm=jpg&fl=progressive&q=90) no-repeat center",
                                                            "backgroundSize": "cover",
                                                          }
                                    }
                        />
                        <Unknown>
                                    
                        </Unknown>
            </div>
</div>,
        ],
        "className": "",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "breadcrumbs": Array [
              Object {
                "label": "Order Online",
                "to": "/order-online",
              },
            ],
            "className": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="aspect-4-3 my2 col-12 md-hide lg-hide"
                style={
                                Object {
                                                "background": "url(?w=900&fm=jpg&fl=progressive&q=90) no-repeat center",
                                                "backgroundSize": "cover",
                                              }
                }
/>,
              <div>
                details
</div>,
              <div
                className="col-12 md-col-6"
>
                <div
                                className="aspect-4-3 my2 xs-hide sm-hide"
                                style={
                                                Object {
                                                                "background": "url(?w=900&fm=jpg&fl=progressive&q=90) no-repeat center",
                                                                "backgroundSize": "cover",
                                                              }
                                }
                />
                <Unknown>
                                
                </Unknown>
</div>,
            ],
            "className": "flex",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "aspect-4-3 my2 col-12 md-hide lg-hide",
                "style": Object {
                  "background": "url(?w=900&fm=jpg&fl=progressive&q=90) no-repeat center",
                  "backgroundSize": "cover",
                },
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "details",
              },
              "ref": null,
              "rendered": "details",
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <div
                    className="aspect-4-3 my2 xs-hide sm-hide"
                    style={
                                        Object {
                                                            "background": "url(?w=900&fm=jpg&fl=progressive&q=90) no-repeat center",
                                                            "backgroundSize": "cover",
                                                          }
                    }
/>,
                  <Unknown>
                    
</Unknown>,
                ],
                "className": "col-12 md-col-6",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "aspect-4-3 my2 xs-hide sm-hide",
                    "style": Object {
                      "background": "url(?w=900&fm=jpg&fl=progressive&q=90) no-repeat center",
                      "backgroundSize": "cover",
                    },
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "children": Array [],
                  },
                  "ref": null,
                  "rendered": Array [],
                  "type": Symbol(react.fragment),
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
